sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/polling/OperationLocationPollingStrategy.java

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.util.polling;
+
+import com.azure.core.exception.AzureException;
+import com.azure.core.experimental.util.polling.implementation.DefaultJsonSerializer;
+import com.azure.core.experimental.util.polling.implementation.ImplUtils;
+import com.azure.core.experimental.util.polling.implementation.PollingConstants;
+import com.azure.core.experimental.util.polling.implementation.PollingUtils;
+import com.azure.core.experimental.util.polling.implementation.PostPollResult;
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.OperationResourcePollingStrategy;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollingContext;
+import com.azure.core.util.polling.PollingStrategyOptions;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+/**
+ * Implements an operation location polling strategy, from Operation-Location.
+ *
+ * @param <T> the type of the response type from a polling call, or BinaryData if raw response body should be kept
+ * @param <U> the type of the final result object to deserialize into, or BinaryData if raw response body should be
+ *           kept
+ */
+public class OperationLocationPollingStrategy<T, U> extends OperationResourcePollingStrategy<T, U> {
```

Comment:
Consider implementing the PollingStrategy directly instead of extending from `OperationResourcePollingStrategy` as they are two separate strategies and can evolve independently.

Reply:
Could we delay this to the time we move the class to core?

Currently a full implementation without `OperationResourcePollingStrategy` would involve more refactor/exposure from azure-core,
E.g. it would need to call a package private `operationResourceCanPoll` method
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollingUtil.java#L174-L188

Reply:
Yes, we can defer this change to a later release as this is in experimental package. We can go ahead with what you currently have. We can revisit this discussion before moving this to azure-core.